site/src/Map.jsx

```diff
@@ -1,135 +1,313 @@
-  const onMouseLeave = useCallback(() => setCursor('auto'), []);
+  const onMouseEnter = useCallback(() => setCursor("pointer"), []);
+  const onMouseLeave = useCallback(() => setCursor("auto"), []);
 
-  const onClick = useCallback(event => {
+  const onClick = useCallback((event) => {
     const feature = event.features && event.features[0];
-
     if (feature) {
-      setMapEntity(event.features[0].properties);
+      setMapEntity({
+        theme: feature.source,
+        type: feature.sourceLayer,
+        ...feature.properties,
+      });
     } else {
       setMapEntity({});
     }
   }, []);
 
-  function getMapStyle() {
-    let mapStyle;
-
-    if (pmTilesReady){
-     mapStyle =  mode === 'theme-light' ? MAP_STYLE : DARK_MAP_STYLE;
-    } else {
-      mapStyle = undefined;
-    }
-    return mapStyle;
-  }
-
   return (
     <>
       <div className={`map ${mode}`}>
         <MapLibreMap
-          id="myMap"
+          ref={mapRef}
           onMouseEnter={onMouseEnter}
           onMouseLeave={onMouseLeave}
+          onLoad={syncInteractiveLayerIds}
           onClick={onClick}
           cursor={cursor}
           hash={true}
+          mapStyle={MAP_STYLE}
           interactiveLayerIds={interactiveLayerIds}
           initialViewState={INITIAL_VIEW_STATE}
-          mapStyle={getMapStyle()}
-          style={{position: 'fixed', width: '100%', height: 'calc(100vh - 60px)'}}
+          style={{
+            position: "fixed",
+            width: "100%",
+            height: "calc(100vh - 60px)",
+          }}
+          attributionControl={false}
         >
-          <Source id="overture-places" type="vector" url={PLACES_PMTILES_URL}>
-            <Layer {...PLACES_MAP_STYLE} layout={{visibility: interactiveLayerIds.includes('places') ? 'visible' : 'none'}} />
-          </Source>
-          <NavigationControl position='top-right'></NavigationControl>
+          <ThemeSource name="base" url={PMTILES_URL} />
+          <ThemeSource name="buildings" url={PMTILES_URL} />
+          <ThemeSource name="places" url={PMTILES_URL} />
+          <ThemeSource name="divisions" url={PMTILES_URL} />
+
+          <Layer
+            id="divisions_division"
+            type="symbol"
+            source="divisions"
+            source-layer="division"
+            paint={{
+              "text-color": mode === "theme-light" ? "black" : "white",
+              "text-halo-color": mode === "theme-light" ? "white" : "black",
+              "text-halo-width": 1,
+            }}
+            layout={{
+              "text-font": ["Noto Sans Regular"],
+              "text-field": ["get", "@name"],
+              "text-size": 11,
+            }}
+          />
+
+          {visibleThemes.includes("base") ? (
+            <>
+              <ThemeTypeLayer
+                theme="base"
+                type="land"
+                point
+                line
+                polygon
+                color="#ccebc5"
+              />
```

Comment:
What are your thoughts on generating these layers more programmatically?

Long term, I was thinking that we generate a manifest for each Overture release which was used to dynamically generated the sources/layers.